crates/bevy_ecs/macros/src/lib.rs

```diff
@@ -45,9 +45,9 @@ impl Parse for AllTuples {
 #[proc_macro]
 pub fn all_tuples(input: TokenStream) -> TokenStream {
     let input = parse_macro_input!(input as AllTuples);
-    let len = (input.start..=input.end).count();
+    let len = (0..=input.end).count();
```

Comment:
What's the purpose of this change?
Wouldn't this break other uses of `all_tuples!`?

Reply:
Without this change, the slicing on line 68 panics during compilation. The changed region here only generates the tuple types that are then filtered on line 68. Since SystemGroup is really only valid for tuples of size 2 and higher, it failed to compile without this change.